crates/toolchain/transpiler/src/elf.rs

```diff
@@ -93,18 +98,45 @@ impl Elf {
         #[cfg(feature = "function-span")]
         {
             if let Some((symtab, stringtab)) = elf.symbol_table()? {
+                let mut fn_names = Vec::new();
+                for symbol in symtab.iter() {
+                    if symbol.st_symtype() == elf::abi::STT_FUNC {
+                        let raw_name = stringtab.get(symbol.st_name as usize).unwrap().to_string();
+                        let demangled_name = rustc_demangle::demangle(&raw_name).to_string();
+                        fn_names.push((demangled_name, symbol.st_name));
+                    }
+                }
+
+                let mut buf = Vec::new();
+                let mut offsets = HashMap::new();
+                buf.push(0);
+                for (name, st_name) in fn_names {
+                    if let Entry::Vacant(e) = offsets.entry(st_name) {
+                        let offset = buf.len();
+                        e.insert(offset);
+                        buf.extend_from_slice(name.as_bytes());
+                        buf.push(0);
+                    }
+                }
+
                 for symbol in symtab.iter() {
                     if symbol.st_symtype() == elf::abi::STT_FUNC {
                         fn_bounds.insert(
                             symbol.st_value as u32,
                             FnBound {
                                 start: symbol.st_value as u32,
                                 end: (symbol.st_value + symbol.st_size - (WORD_SIZE as u64)) as u32,
-                                name: stringtab.get(symbol.st_name as usize).unwrap().to_string(),
+                                name: offsets[&symbol.st_name].to_string(),
                             },
                         );
                     }
                 }
+
+                let guest_symbols_file = std::env::var("GUEST_SYMBOLS_PATH")
+                    .map(|path| std::fs::File::create(path).unwrap());
```

Comment:
let's not unwrap and propagate the error; also make sure the error displays the path if case it fails to create (this can happen eg if a directory doesn't exist)